### PR TITLE
Parameter overriding

### PR DESCRIPTION
**Pull request by jihoonl**
- export tag extractor

https://github.com/robotics-in-concert/rocon_concert/pull/196

### CODE REVIEW

File: rocon_interactions/src/rocon_interactions/interactions.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def load_msgs_from_yaml_resource(resource_name):
     '''
     interactions = []
     try:
+        print(str(resource_name))
```
Comment (stonier) [inline rocon_interactions/src/rocon_interactions/interactions.py]: This should disappear.

File: rocon_python_utils/src/rocon_python_utils/ros/resources.py [hunk 2]
```diff
@@ -71,3 +74,35 @@ def find_resource(package, filename, rospack=None):
     except rospkg.ResourceNotFound:
         raise rospkg.ResourceNotFound("[%s] is not a package or launch file name" % package)
     return None
+
+
+def extract_tags_from_packages(export_tag): 
+    '''
+      Scans the package path looking for exports and grab the ones we are interested in.
+      
+      @param export : export tagname
```
Comment (stonier) [inline rocon_python_utils/src/rocon_python_utils/ros/resources.py]: export -> export_tag

File: rocon_python_utils/src/rocon_python_utils/ros/resources.py [hunk 3]
```diff
@@ -71,3 +74,35 @@ def find_resource(package, filename, rospack=None):
...
+                filename_relative_path = export.content
+                resource_name = package.name + '/' + os.path.splitext(os.path.basename(filename_relative_path))[0]
+                resource_filename = os.path.join(os.path.dirname(package.filename), filename_relative_path)
+                if not os.path.isfile(resource_filename):
+                    rospy.logwarn("couldn't find definition for exported resource [%s]" % resource_name)
+                resources[resource_name] = resource_filename
+    return resources
+
+
+def _get_package_path():
```
Comment (stonier) [inline rocon_python_utils/src/rocon_python_utils/ros/resources.py]: This is not getting a package path - it is getting an index. Strange name.

File: rocon_python_utils/src/rocon_python_utils/ros/resources.py [hunk 4]
```diff
@@ -71,3 +74,35 @@ def find_resource(package, filename, rospack=None):
...
+    
+    package_index = _get_package_path()
+    resources = {}
+    for package in package_index.values():
+        for export in package.exports:
+            if export.tagname == export_tag:
+                filename_relative_path = export.content
+                resource_name = package.name + '/' + os.path.splitext(os.path.basename(filename_relative_path))[0]
+                resource_filename = os.path.join(os.path.dirname(package.filename), filename_relative_path)
+                if not os.path.isfile(resource_filename):
```
Comment (stonier) [inline rocon_python_utils/src/rocon_python_utils/ros/resources.py:97]: Not sure about this error handling - it's a library module, so I typically avoid logwarn's in these and use exceptions to float the error up so you can let the user deal with it as they wish. But this is not a fatal exception and you still want to return a result even if there were a few errors.

Here the logging doesn't help the calling program know what failed...only way to know is have eyeballs on the output. For this I sometimes return a tuple, e.g. (resource_index, invalid_resources). Something like this?

File: rocon_python_utils/src/rocon_python_utils/ros/resources.py [hunk 5]
```diff
@@ -71,3 +74,35 @@ def find_resource(package, filename, rospack=None):
     except rospkg.ResourceNotFound:
         raise rospkg.ResourceNotFound("[%s] is not a package or launch file name" % package)
     return None
+
+
+def extract_tags_from_packages(export_tag):
```
Comment (stonier) [inline rocon_python_utils/src/rocon_python_utils/ros/resources.py]: Strange name...doesn't say anything about resources. `resource_index_from_package_exports` might be better.

I'd put a second argument `ros_package_path` (just like the package_index_from_package_path) in here. That's a habit I picked up watching what Damon and Dirk and William do - it lets them write tests on small slices of the package or even a fake test package path.

**Comment by stonier** [comment]
Some update jobs for you to do...otherwise looks good.

File: rocon_python_utils/src/rocon_python_utils/ros/resources.py (hunk 4 above)
Comment (jihoonl) [inline rocon_python_utils/src/rocon_python_utils/ros/resources.py:97]: That would be nicer way. I will do that

**Comment by jihoonl** [comment]
Updated.

File: rocon_python_utils/src/rocon_python_utils/ros/resources.py [hunk 6]
```diff
@@ -71,3 +74,37 @@ def find_resource(package, filename, rospack=None):
...
+                if not os.path.isfile(resource_filename):
+                    invalid_resources[resource_name] = resource_filename
+                else:
+                    resources[resource_name] = resource_filename
+    return (resources, invalid_resources)
+
+
+def _get_package_index(package_paths):
+    ros_package_path = package_paths if package_paths else os.getenv('ROS_PACKAGE_PATH', '')
+    print(str(ros_package_path))
```
Comment (stonier) [inline rocon_python_utils/src/rocon_python_utils/ros/resources.py]: Drop this print as well.

File: rocon_python_utils/src/rocon_python_utils/ros/resources.py [hunk 7]
```diff
@@ -11,9 +11,12 @@
 import os
 
 # ros
+import rospy
```
Comment (stonier) [inline rocon_python_utils/src/rocon_python_utils/ros/resources.py]: Don't think you need this rospy anymore either if the rospy logging isn't in.

**Comment by jihoonl** [comment]
updated.

**Comment by stonier** [comment]
:+1:

**Pull request merged**